src/modules/auth/auth.controller.ts

```diff
@@ -162,4 +165,93 @@ export class AuthController extends Controller {
       data: { message: "Forgot password email send successfully" },
     });
   }
+
+  @Post("/resetPassword")
+  @Middlewares(validateData(authValidations.resetPass))
+  public async resetPassword(
+    @Body() body: AuthResetPass
+  ): Promise<APIResponse<Message>> {
+    const { password, ...restOfBody } = body;
+
+    const verified = await otpService.verify(restOfBody);
+    if (!verified) {
+      this.setStatus(statusConst.unAuthenticated.code);
+      return toResponse({ error: statusConst.unAuthenticated.message });
+    }
+
+    const newPassword = await authSerivce.hashPassword(password);
+
+    await prisma.users.update({
+      where: { email: body.email },
+      data: { password_hash: newPassword },
+    });
+
+    return toResponse({
+      data: { message: "Password has been reset successfully" },
+    });
+  }
+
+  @Post("/changePassword")
+  @Middlewares(validateData(authValidations.changePass))
+  public async changePassword(
+    @Body() body: AuthChangePass
+  ): Promise<APIResponse<Message>> {
+    const { email, password } = body;
+
+    const newHashedPassword = await authSerivce.hashPassword(password);
+
+    await prisma.users.update({
+      where: {
+        email: email,
+      },
+      data: {
+        password_hash: newHashedPassword,
+      },
+    });
+
+    return toResponse({
+      data: { message: "Password has been changed  successfully" },
+    });
+  }
+
+  @Post("/resendVerification")
+  @Middlewares(validateData(authValidations.resendVerification))
+  public async resendVerification(
+    @Body() body: AuthVerification
+  ): Promise<APIResponse<Message>> {
+    const user = await userService.fetchByEmail(body.email);
+    if (!user) {
+      this.setStatus(statusConst.notFound.code);
+      return toResponse({ error: statusConst.notFound.message });
+    }
+
+    await otpService.send(user, body.emailType);
```

Comment:
we shouldn't take template type in body. We already know ifs for sending the verification email again, so template type should be fixed here.